Fuse/Classes/Fuse.swift

```diff
@@ -303,7 +304,9 @@ extension Fuse {
             self.searchQueue.async {
```

Comment:
We're able to hit this as well - it's because the `searchQueue` is concurrent, so if the `aList.count > chunkSize`, it will be susceptible to crashing during search. This can be proven by changing `chunkSize` to be >= `aList.count`. We no longer can produce the crash when that's the case. 

I think the need for `NSLock` could be avoided if we just change this line to:

`self.searchQueue.async(group: group, qos: .userInteractive, flags: .barrier, execute: { ... })`

(and remove the `group.enter()` and `group.leave()` as this will do it for us)

I haven't tested the performance of this vs. using an `NSLock`, but I think it's cleaner and should have at least similar performance.

Reply:
I think that submitting each block as `.barrier` would effectively serialize the queue.

Reply:
I suppose that instead of locking, what we could do is have each block append to its very own array, then when all are done, consolidate these arrays into a single `items` array.

Reply:
Ooh. I can't immediately envision how that would look but that sounds cool.

I think that https://github.com/krisk/fuse-swift/pull/43 might be a way to avoid any locking too.